source/git/gitsource_unix.go

```diff
@@ -4,12 +4,21 @@ package git
 
 import (
 	"context"
+	"fmt"
 	"os/exec"
+	"strings"
 	"syscall"
 	"time"
 )
 
+const defaultUmask = "0022"
+
 func runProcessGroup(ctx context.Context, cmd *exec.Cmd) error {
+	// Dirty workaround to use standard user umask for git process.
```

Comment:
I think this is too ugly for even my taste :) In addition to shell dependency, this can easily go wrong with signals and args that contain spaces.

Reply:
Using reexec now.